File: Maximum Average Subarray I - Leetcode 643/brute force.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """

        def avg(vals):
            return sum(vals) / float(len(vals))

        # Note, setting max_avg to any constant numeric value 
        # here is incorrect!!! 
        # Say 0, if nums = [-1, -2, -3, -4], k = 2, 
        # the output would be incorrect
        max_avg = avg(nums[:k])
        for i in range(len(nums) - k + 1):
            max_avg = max(max_avg, avg(nums[i:i+k])) # avg is O(k)
        return max_avg

File: Maximum Average Subarray I - Leetcode 643/brute force.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double max_avg = 0;

        // setting max_avg to any constant numeric value 
        // here is incorrect!!! 
        // So initialize it with some values from the input
        for (int i = 0; i < k; ++i)
            max_avg += nums[i];
        max_avg /= k;

        for (int i = 0; i <= nums.length - k; ++i) {
            double avg = 0;
            for (int j = i; j < i + k; ++j)
                avg += nums[j];
            avg /= k;
            max_avg = Math.max(max_avg, avg); 
        }
        return max_avg;

    }
}
